.github/workflows/ci.yml

```diff
@@ -79,3 +83,118 @@ jobs:
     steps:
       - name: Run Codecov 
         run: bash <(curl -s https://codecov.io/bash)
+
+  # Increments to the next tag on master only, bumps version based on the recent commits
+  new_tag:
+    if: (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master') && github.event_name == 'push'
+    runs-on: ubuntu-latest
+    needs: test
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Set up Go
+        uses: actions/setup-go@v5
+        with:
+          go-version: '1.22'
+      - name: Get next version
+        id: new_git_tag
+        run: |
+          git remote add upstream https://github.com/gocolly/colly || true
+          git fetch upstream --tags
+          git tag # This is just for future debug, lists the past tags
+          go install github.com/caarlos0/svu@latest
+          NEW_TAG=$(svu next --tag-mode all-branches --force-patch-increment)
+          echo "Using the next tag of ${NEW_TAG} for this release"
+          echo "new_tag=$NEW_TAG" >> $GITHUB_ENV
+          echo "new_tag=$NEW_TAG" >> $GITHUB_OUTPUT
+    outputs:
+      new_tag: ${{ steps.new_git_tag.outputs.new_tag }}
+
+  # Creates a new release page on GitHub when master changes
+  create_release:
+    if: (github.ref == 'refs/heads/main' || github.ref == 'refs/heads/master') && github.event_name == 'push'
+    runs-on: ubuntu-latest
+    needs: new_tag
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Create Release
+        id: create_release
+        uses: actions/create-release@v1
```

Comment:
This action seems to be unmaintained, I wouldn't use it: https://github.com/actions/create-release

Reply:
Maybe try this fork instead for now: https://github.com/marketplace/actions/gh-release